### ISSUE TITLE
Add support for the latest Pydantic versions

### ISSUE THREAD

**Issue by yanbo-huang** [redis/redis-om-python#595]
The current `redis-om-python`'s dependency constraint for Pydantic version is `pydantic = ">=1.10.2,<2.1.0"`.
ATM, the latest Pydantic version is `v2.6.1`. Its development pace is quite fast. Within one year, lots of features are added and bugs are fixed.
Could you please add support for a higher version of Pydantic? Then our projects don't need to hold onto the lower versions.

**Comment by yanbo-huang** [comment]
I went through all chats in the pull requests and issues. Noticed there are already a few PRs updating the Pydantic's version.
They are still waiting for approval for quite a long time. We are in a hurry to migrate to the latest Pydantic version in our project, so I created a fork and updated the version to `pydantic = ">=1.10.2,<3.0.0"`.  I tested it in my own environments, it is working fine so far. For people in the same situation, feel free to use it:
```shell
# put this in your package management file (eg, `pyproject.toml`)
redis-om = {git = "https://github.com/yanbo-huang/redis-om-python.git"}
```

**Comment by prutheus** [comment]
what is the current status on this? @slorello89 @uglide

**Comment by gaby** [comment]
Fixed by #597

**Comment by banker** [comment]
Thanks, @gaby!
